cdap-app-fabric/src/test/java/co/cask/cdap/AllProgramsApp.java

```diff
@@ -245,17 +268,27 @@ public void run(DatasetContext context) throws Exception {
   public static class NoOpService extends AbstractService {
 
     public static final String NAME = "NoOpService";
+    public static final String ENDPOINT = "no-op";
 
     @Override
     protected void configure() {
       addHandler(new NoOpHandler());
     }
 
-    private class NoOpHandler extends AbstractHttpServiceHandler {
+    public class NoOpHandler extends AbstractHttpServiceHandler {
       @Override
       protected void configure() {
         useDatasets(DATASET_NAME);
       }
+
+      @Path(ENDPOINT)
+      @GET
+      public void handler(HttpServiceRequest request, HttpServiceResponder responder) {
+        LOG.info("Endpoint {} called in service {}", ENDPOINT, NAME);
+        KeyValueTable table = getContext().getDataset(DATASET_NAME);
```

Comment:
We would then have to use a different dataset for every program. I don't think having a single dataset should matter much, because the access is recorded while the dataset is being retrieved. If an incorrect dataset gets recorded, then the same incorrect dataset would get returned and all tests would fail. What do you say?